### PR TITLE
Crud book

### PR DESCRIPTION
**Pull request by joredudiaz**
## Summary

- Creates a book
- Updates a book
- List books
- Shows a book searched by Id
- Deletes a book

## Jira Story

https://trello.com/c/0aqhg7md

### CODE REVIEW

**Review by Jimenarosello — APPROVED** [review]
Looks good to me 😄

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -1,18 +1,69 @@
...
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public Book findOne(@PathVariable long id) throws BookNotFoundException {
+        return bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
```
Comment (darkBasic666) [inline src/main/java/wolox/training/controllers/BookController.java:66]: response 201

**Review by darkBasic666 — CHANGES_REQUESTED** [review]
Add javadocs

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -1,18 +1,69 @@
...
+    public Book create(@RequestBody Book book) {
+        return bookRepository.save(book);
+    }
+
+    @DeleteMapping("/{id}")
```
Comment (darkBasic666) [inline src/main/java/wolox/training/controllers/BookController.java:77]: return 204

File: src/main/java/wolox/training/controllers/BookController.java [hunk 3]
```diff
@@ -1,18 +1,69 @@
...
+    public void delete(@PathVariable Long id) throws BookNotFoundException {
+        bookRepository.findById(id)
+            .orElseThrow(BookNotFoundException::new);
+        bookRepository.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
```
Comment (darkBasic666) [inline src/main/java/wolox/training/controllers/BookController.java:93]: return 204

Comment (darkBasic666) [reply src/main/java/wolox/training/controllers/BookController.java:93]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: src/main/java/wolox/training/controllers/BookController.java (hunk 1 above)
Comment (joredudiaz) [reply src/main/java/wolox/training/controllers/BookController.java:66]: HttpStatus.CREATED is 201 status response

**Review by darkBasic666 — APPROVED** [review]

**Comment by darkBasic666** [comment]
Maybe add a integral test, using db, services, etc, help to test entire flow.
If you do, use a groove routine for clean DB, and utils for populate.

**Pull request merged**